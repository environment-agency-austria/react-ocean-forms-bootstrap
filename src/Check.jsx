/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faSpinner from '@fortawesome/fontawesome-free-solid/faSpinner';
import faExclamationCircle from '@fortawesome/fontawesome-free-solid/faExclamationCircle';
import {
  Input,
  FormGroup,
  Label,
  Col,
  Button,
  Alert,
  InputGroup,
  InputGroupAddon,
} from 'reactstrap';
import { fieldMetaShape, fieldShape } from 'react-ocean-forms';

import { BaseFieldError } from './FieldError';

/**
 * Component for displaying bootstrap
 * form groups with an html checkbox and
 * oForm support
 */
class Check extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      infoVisible: false,
    };

    this.handleChange = this.handleChange.bind(this);
    this.toggleInfo = this.toggleInfo.bind(this);
  }

  /**
   * Toggles the visibility state of the info alert
   */
  toggleInfo() {
    this.setState(prevState => ({
      infoVisible: !prevState.infoVisible,
    }));
  }

  /**
   * Manually handle the onChange event of
   * the checkbox so we can use the checked
   * property instead of value.
   */
  handleChange(event) {
    const { field } = this.props;
    field.onChange({
      target: {
        name: field.name,
        value: event.target.checked,
      },
    });
  }

  render() {
    const {
      field,
      label,
      className,
      meta,
      info,
      field: {
        value,
      },
    } = this.props;

    const { infoVisible } = this.state;

    const isChecked = value === true;
    const inputGroupClass = info !== undefined ? 'has-info' : undefined;
    const groupClass = (className + (meta.valid ? '' : ' is-invalid')).trim();
    const labelString = meta.stringFormatter(label);

    return (
      <FormGroup row className={groupClass}>
        <Col sm={3} className="text-right check-label-col">
          {!meta.valid && <FontAwesomeIcon icon={faExclamationCircle} className="mr-2" />}
          {meta.isValidating && <FontAwesomeIcon icon={faSpinner} spin className="mr-2" />}
        </Col>
        <Col sm={9}>
          <InputGroup className={inputGroupClass}>
            <Label check>
              {!meta.plaintext && (
                <Input
                  id={field.id}
                  name={field.name}
                  type="checkbox"
                  onBlur={field.onBlur}
                  invalid={!meta.valid}
                  defaultChecked={isChecked}
                  onClick={this.handleChange}
                  disabled={field.disabled}
                />
              )}
              {labelString}
              <BaseFieldError
                id={`${field.id}_errors`}
                invalid={!meta.valid}
                error={meta.error}
                context={{ stringFormatter: meta.stringFormatter }}
              />
            </Label>
            {info && (
              <InputGroupAddon addonType="append">
                <Button onClick={this.toggleInfo} outline>
                  <FontAwesomeIcon icon={faExclamationCircle} />
                </Button>
              </InputGroupAddon>
            )}
          </InputGroup>

          {info && (
            <Alert
              color="success"
              className="mt-2"
              isOpen={infoVisible}
              toggle={this.toggleInfo}
            >
              {meta.stringFormatter(info)}
            </Alert>
          )}
        </Col>
      </FormGroup>
    );
  }
}

Check.defaultProps = {
  info: undefined,
  className: undefined,
};

Check.propTypes = {
  label: PropTypes.string.isRequired,
  info: PropTypes.string,
  meta: fieldMetaShape.isRequired,
  field: fieldShape.isRequired,
  className: PropTypes.string,
};

export default Check;
