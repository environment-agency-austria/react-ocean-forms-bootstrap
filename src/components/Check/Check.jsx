/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import {
  Input,
  Label,
  Col,
  InputGroup,
} from 'reactstrap';
import { FormText } from 'react-ocean-forms';
import { fieldMetaShape, fieldShape } from 'react-ocean-forms-legacy';

import { FieldRow } from '../FieldLine/FieldRow';
import InvalidAlert from '../FieldLine/InvalidAlert';
import ValidatingSpinner from '../FieldLine/ValidatingSpinner';
import { InfoAddonButton } from '../FieldLine/InfoAddonButton';
import InfoAlert from '../FieldLine/InfoAlert';
import FieldError from '../FieldLine/FieldError';

/**
 * Component for displaying bootstrap
 * form groups with an html checkbox and
 * oForm support
 */
class Check extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      infoVisible: false,
    };

    this.handleChange = this.handleChange.bind(this);
    this.toggleInfo = this.toggleInfo.bind(this);
  }

  /**
   * Toggles the visibility state of the info alert
   */
  toggleInfo() {
    this.setState(prevState => ({
      infoVisible: !prevState.infoVisible,
    }));
  }

  /**
   * Manually handle the onChange event of
   * the checkbox so we can use the checked
   * property instead of value.
   */
  handleChange(event) {
    const { field } = this.props;
    field.onChange({
      target: {
        name: field.name,
        value: event.target.checked,
      },
    });
  }

  render() {
    const {
      field,
      label,
      className,
      meta,
      info,
      field: {
        value,
      },
    } = this.props;

    const { infoVisible } = this.state;

    const isChecked = value === true;
    const inputGroupClass = info !== undefined ? 'has-info' : undefined;

    return (
      <FieldRow meta={meta} className={className}>
        <Col sm={3} className="text-right check-label-col">
          <InvalidAlert valid={meta.valid} />
          <ValidatingSpinner isValidating={meta.isValidating} />
        </Col>
        <Col sm={9}>
          <InputGroup className={inputGroupClass}>
            <Label check>
              {!meta.plaintext && (
                <Input
                  id={field.id}
                  name={field.name}
                  type="checkbox"
                  onBlur={field.onBlur}
                  invalid={!meta.valid}
                  defaultChecked={isChecked}
                  onClick={this.handleChange}
                  disabled={field.disabled}
                />
              )}
              <FormText text={label} />
              <FieldError
                id={`${field.id}_errors`}
                invalid={!meta.valid}
                error={meta.error}
              />
            </Label>
            <InfoAddonButton info={info} plaintext={meta.plaintext} onClick={this.toggleInfo} />
          </InputGroup>
          <InfoAlert
            visible={infoVisible}
            info={info}
            plaintext={meta.plaintext}
            onClose={this.toggleInfo}
          />
        </Col>
      </FieldRow>
    );
  }
}

Check.displayName = 'Check';

Check.defaultProps = {
  info: undefined,
  className: undefined,
};

Check.propTypes = {
  label: PropTypes.string.isRequired,
  info: PropTypes.string,
  meta: fieldMetaShape.isRequired,
  field: fieldShape.isRequired,
  className: PropTypes.string,
};

export default Check;
