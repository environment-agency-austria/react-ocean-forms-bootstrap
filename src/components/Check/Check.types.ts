import { IFieldComponentProps } from 'react-ocean-forms';
import { IInfoProps } from '../FieldLine';

/**
 * Props for the Check component
 */
export interface ICheckProps extends IFieldComponentProps, IInfoProps {
  /**
   * CSS class name
   */
  className?: string;
}
