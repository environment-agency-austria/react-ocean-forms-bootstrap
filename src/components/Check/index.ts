export * from './Check';
export * from './Check.types';
