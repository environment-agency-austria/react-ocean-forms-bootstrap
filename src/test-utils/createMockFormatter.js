export default () => jest.fn().mockImplementation(id => id);
