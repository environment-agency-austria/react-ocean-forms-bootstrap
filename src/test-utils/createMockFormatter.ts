export const createMockFormatter = (): jest.Mock => jest.fn().mockImplementation((id: string) => id);
