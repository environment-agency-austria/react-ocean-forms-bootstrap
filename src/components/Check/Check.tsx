/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as React from 'react';

import { FormText, withField } from 'react-ocean-forms';
import { Col, Input, InputGroup, Label } from 'reactstrap';

import { FieldError } from '../FieldLine/FieldError';
import { FieldRow } from '../FieldLine/FieldRow';
import { InfoAddonButton } from '../FieldLine/InfoAddonButton';
import { InfoAlert } from '../FieldLine/InfoAlert';
import { InvalidAlert } from '../FieldLine/InvalidAlert';
import { ValidatingSpinner } from '../FieldLine/ValidatingSpinner';

import { ICheckProps } from './Check.types';

interface ICheckState {
  infoVisible: boolean;
}

/**
 * Component for displaying bootstrap
 * form groups with an html checkbox and
 * oForm support
 */
export class BaseCheck extends React.Component<ICheckProps, ICheckState> {
  public static displayName: string = 'Check';

  constructor(props: ICheckProps) {
    super(props);

    this.state = {
      infoVisible: false,
    };
  }

  /**
   * Toggles the visibility state of the info alert
   */
  private toggleInfo = (): void => {
    this.setState(prevState => ({
      infoVisible: !prevState.infoVisible,
    }));
  }

  /**
   * Manually handle the onChange event of
   * the checkbox so we can use the checked
   * property instead of value.
   */
  private handleChange = (event: React.MouseEvent<HTMLInputElement>): void => {
    const { field } = this.props;

    // event.target.checked exists in this case
    // because of the HtmlInputElement
    // @ts-ignore
    const checked = event.target.checked as boolean;

    field.onChange({
      target: {
        value: checked,
      },
    });
  }

  public render(): JSX.Element {
    const {
      field,
      label,
      className,
      meta,
      info,
      field: {
        value,
      },
    } = this.props;

    const { infoVisible } = this.state;

    const isChecked = value === true;
    const inputGroupClass = info !== undefined ? 'has-info' : undefined;
    const disabled = field.disabled || meta.plaintext;

    return (
      <FieldRow meta={meta} className={className}>
        <Col sm={3} className="text-right check-label-col">
          <InvalidAlert valid={meta.valid} />
          <ValidatingSpinner isValidating={meta.isValidating} />
        </Col>
        <Col sm={9}>
          <InputGroup className={inputGroupClass}>
            <Label check>
              <Input
                id={field.id}
                name={field.name}
                type="checkbox"
                onBlur={field.onBlur}
                invalid={!meta.valid}
                checked={isChecked}
                onClick={this.handleChange}
                disabled={disabled}
              />
              <FormText text={label} />
              <FieldError
                id={`${field.id}_errors`}
                invalid={!meta.valid}
                error={meta.error}
              />
            </Label>
            <InfoAddonButton info={info} plaintext={meta.plaintext} onClick={this.toggleInfo} />
          </InputGroup>
          <InfoAlert
            visible={infoVisible}
            info={info}
            plaintext={meta.plaintext}
            onClose={this.toggleInfo}
          />
        </Col>
      </FieldRow>
    );
  }
}

export const Check = withField(BaseCheck);
