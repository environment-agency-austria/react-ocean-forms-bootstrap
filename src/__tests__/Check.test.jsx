import React from 'react';
import { shallow } from 'enzyme';

import createMockFormatter from '../test-utils/createMockFormatter';
import Check from '../Check';

describe('<Check />', () => {
  const FIELD_ID = 'field0';
  const FIELD_NAME = 'field0';
  const FIELD_LABEL = 'field0';

  const meta = {
    valid: true,
    error: undefined,
    isValidating: undefined,
    stringFormatter: createMockFormatter(),
  };
  const field = {
    value: true,
    invalid: false,
    id: FIELD_ID,
    name: FIELD_NAME,
    disabled: false,
    onChange: jest.fn(),
    onBlur: jest.fn(),
  };

  const setup = props => shallow((
    <Check
      label={FIELD_LABEL}
      meta={meta}
      field={field}
      {...props}
    />
  ));

  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call field.onChange when the input changes', () => {
    const wrapper = setup();
    const event = { target: { name: FIELD_ID, checked: false } };

    wrapper.find('Input').simulate('click', event);
    expect(field.onChange).toHaveBeenCalledWith({
      target: {
        name: FIELD_ID,
        value: false,
      },
    });
  });
});
