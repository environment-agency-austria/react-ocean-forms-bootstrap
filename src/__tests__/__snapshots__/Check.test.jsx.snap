// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Check /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Check
    field={
      Object {
        "id": "field0",
        "invalid": false,
        "name": "field0",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": true,
      }
    }
    label="field0"
    meta={
      Object {
        "error": undefined,
        "isValidating": undefined,
        "stringFormatter": [MockFunction] {
          "calls": Array [
            Array [
              "field0",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": "field0",
            },
          ],
        },
        "valid": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Col
          className="text-right check-label-col"
          sm={3}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        />,
        <Col
          sm={9}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <InputGroup
            tag="div"
          >
            <Label
              check={true}
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Input
                defaultChecked={true}
                id="field0"
                invalid={false}
                name="field0"
                onBlur={[MockFunction]}
                onClick={[Function]}
                type="checkbox"
              />
              field0
              <FieldError
                context={
                  Object {
                    "stringFormatter": [MockFunction] {
                      "calls": Array [
                        Array [
                          "field0",
                        ],
                      ],
                      "results": Array [
                        Object {
                          "isThrow": false,
                          "value": "field0",
                        },
                      ],
                    },
                  }
                }
                error={null}
                id="field0_errors"
                invalid={false}
              />
            </Label>
          </InputGroup>
        </Col>,
      ],
      "className": "undefined",
      "row": true,
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            false,
            undefined,
          ],
          "className": "text-right check-label-col",
          "sm": 3,
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          undefined,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <InputGroup
              tag="div"
            >
              <Label
                check={true}
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  defaultChecked={true}
                  id="field0"
                  invalid={false}
                  name="field0"
                  onBlur={[MockFunction]}
                  onClick={[Function]}
                  type="checkbox"
                />
                field0
                <FieldError
                  context={
                    Object {
                      "stringFormatter": [MockFunction] {
                        "calls": Array [
                          Array [
                            "field0",
                          ],
                        ],
                        "results": Array [
                          Object {
                            "isThrow": false,
                            "value": "field0",
                          },
                        ],
                      },
                    }
                  }
                  error={null}
                  id="field0_errors"
                  invalid={false}
                />
              </Label>
            </InputGroup>,
            undefined,
          ],
          "sm": 9,
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Label
                  check={true}
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <Input
                    defaultChecked={true}
                    id="field0"
                    invalid={false}
                    name="field0"
                    onBlur={[MockFunction]}
                    onClick={[Function]}
                    type="checkbox"
                  />
                  field0
                  <FieldError
                    context={
                      Object {
                        "stringFormatter": [MockFunction] {
                          "calls": Array [
                            Array [
                              "field0",
                            ],
                          ],
                          "results": Array [
                            Object {
                              "isThrow": false,
                              "value": "field0",
                            },
                          ],
                        },
                      }
                    }
                    error={null}
                    id="field0_errors"
                    invalid={false}
                  />
                </Label>,
                undefined,
              ],
              "className": undefined,
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "check": true,
                  "children": Array [
                    <Input
                      defaultChecked={true}
                      id="field0"
                      invalid={false}
                      name="field0"
                      onBlur={[MockFunction]}
                      onClick={[Function]}
                      type="checkbox"
                    />,
                    "field0",
                    <FieldError
                      context={
                        Object {
                          "stringFormatter": [MockFunction] {
                            "calls": Array [
                              Array [
                                "field0",
                              ],
                            ],
                            "results": Array [
                              Object {
                                "isThrow": false,
                                "value": "field0",
                              },
                            ],
                          },
                        }
                      }
                      error={null}
                      id="field0_errors"
                      invalid={false}
                    />,
                  ],
                  "tag": "label",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "defaultChecked": true,
                      "id": "field0",
                      "invalid": false,
                      "name": "field0",
                      "onBlur": [MockFunction],
                      "onClick": [Function],
                      "type": "checkbox",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "field0",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "context": Object {
                        "stringFormatter": [MockFunction] {
                          "calls": Array [
                            Array [
                              "field0",
                            ],
                          ],
                          "results": Array [
                            Object {
                              "isThrow": false,
                              "value": "field0",
                            },
                          ],
                        },
                      },
                      "error": null,
                      "id": "field0_errors",
                      "invalid": false,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            className="text-right check-label-col"
            sm={3}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          />,
          <Col
            sm={9}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <InputGroup
              tag="div"
            >
              <Label
                check={true}
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  defaultChecked={true}
                  id="field0"
                  invalid={false}
                  name="field0"
                  onBlur={[MockFunction]}
                  onClick={[Function]}
                  type="checkbox"
                />
                field0
                <FieldError
                  context={
                    Object {
                      "stringFormatter": [MockFunction] {
                        "calls": Array [
                          Array [
                            "field0",
                          ],
                        ],
                        "results": Array [
                          Object {
                            "isThrow": false,
                            "value": "field0",
                          },
                        ],
                      },
                    }
                  }
                  error={null}
                  id="field0_errors"
                  invalid={false}
                />
              </Label>
            </InputGroup>
          </Col>,
        ],
        "className": "undefined",
        "row": true,
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              false,
              undefined,
            ],
            "className": "text-right check-label-col",
            "sm": 3,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <InputGroup
                tag="div"
              >
                <Label
                  check={true}
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <Input
                    defaultChecked={true}
                    id="field0"
                    invalid={false}
                    name="field0"
                    onBlur={[MockFunction]}
                    onClick={[Function]}
                    type="checkbox"
                  />
                  field0
                  <FieldError
                    context={
                      Object {
                        "stringFormatter": [MockFunction] {
                          "calls": Array [
                            Array [
                              "field0",
                            ],
                          ],
                          "results": Array [
                            Object {
                              "isThrow": false,
                              "value": "field0",
                            },
                          ],
                        },
                      }
                    }
                    error={null}
                    id="field0_errors"
                    invalid={false}
                  />
                </Label>
              </InputGroup>,
              undefined,
            ],
            "sm": 9,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Label
                    check={true}
                    tag="label"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <Input
                      defaultChecked={true}
                      id="field0"
                      invalid={false}
                      name="field0"
                      onBlur={[MockFunction]}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    field0
                    <FieldError
                      context={
                        Object {
                          "stringFormatter": [MockFunction] {
                            "calls": Array [
                              Array [
                                "field0",
                              ],
                            ],
                            "results": Array [
                              Object {
                                "isThrow": false,
                                "value": "field0",
                              },
                            ],
                          },
                        }
                      }
                      error={null}
                      id="field0_errors"
                      invalid={false}
                    />
                  </Label>,
                  undefined,
                ],
                "className": undefined,
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "check": true,
                    "children": Array [
                      <Input
                        defaultChecked={true}
                        id="field0"
                        invalid={false}
                        name="field0"
                        onBlur={[MockFunction]}
                        onClick={[Function]}
                        type="checkbox"
                      />,
                      "field0",
                      <FieldError
                        context={
                          Object {
                            "stringFormatter": [MockFunction] {
                              "calls": Array [
                                Array [
                                  "field0",
                                ],
                              ],
                              "results": Array [
                                Object {
                                  "isThrow": false,
                                  "value": "field0",
                                },
                              ],
                            },
                          }
                        }
                        error={null}
                        id="field0_errors"
                        invalid={false}
                      />,
                    ],
                    "tag": "label",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "defaultChecked": true,
                        "id": "field0",
                        "invalid": false,
                        "name": "field0",
                        "onBlur": [MockFunction],
                        "onClick": [Function],
                        "type": "checkbox",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "field0",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "context": Object {
                          "stringFormatter": [MockFunction] {
                            "calls": Array [
                              Array [
                                "field0",
                              ],
                            ],
                            "results": Array [
                              Object {
                                "isThrow": false,
                                "value": "field0",
                              },
                            ],
                          },
                        },
                        "error": null,
                        "id": "field0_errors",
                        "invalid": false,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
            undefined,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
